### PR TITLE
[EC-554] Document process to set up iOS local development

### PR DESCRIPTION
**Pull request by junyuanz1**
## Type of change
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [x] Other

## Objective
I would love to contribute to the Bitwarden project in the future. I have spent several days trying to set up the iOS build environment, and to be honest; it is not as easy as said in the `README.md` file since several steps are required to make this runnable. Therefore, I am trying to add a section to the `CONTRIBUTE.md` file so that people can create the dev environment easily. I think this is a good fit for this section: `https://contributing.bitwarden.com/clients/mobile/#ios-development`

## Code changes
None

## Before you submit
- [ ] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)

### CODE REVIEW

**Comment by eliykat** [comment]
Hi @junyuan-z, thanks for the writeup! We have internal instructions for iOS development, but they rely on having access to the Bitwarden team provisioning profiles & dev certificates, so we haven't made them public. It looks like you've gotten around this by changing the domain and generating your own profiles etc? Is there any other secret sauce in here to get it working without the team profiles?

If you can sign the CLA, I'll see if I can review and incorporate this into our new contributing site, https://contributing.bitwarden.com/.

**Comment by junyuan-z** [comment]
hi @eliykat, that's all the changes that I need to make. and I have signed the CLA but for some reason it still pending.

**Comment by junyuan-z** [comment]
ok, CLA passed (need to make some change to my local git settings)

**Comment by eliykat** [comment]
Thanks! I'm working through some other documentation work at the moment, but will add this to the list to update to contributing.bitwarden.com. I'll leave this PR open until then.

**Comment by MGibson1** [comment]
@eliykat it might be better to hold off for a _little_ bit. I'm almost ready to convert the contributing-docs repo into a public one (🥳 ). I just need to finish up some work to get bitwarden-employee-specific information a little more clearly indicated.

That being said, @junyuan-z, I should be able to attribute you as a co-author at least once I port over these changes to the public repo. Up to you on the route you'd rather go -- direct PR merge or just getting the info out there sooner.

**Comment by eliykat** [comment]
I think @fedemkr was going to look into it, as he knows more about mobile development to assess these instructions/workarounds.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -38,3 +38,218 @@ If you are interested in helping translate the Bitwarden mobile app into another
...
+2. Create an App Group:
+   - Create an App Group by using [Apple Developer Portal](https://developer.apple.com/account/resources/identifiers/list/applicationGroup)
+   - Set the Identifier to `group.com.example.bitwarden`
+3. Create an iCloud Container:
+   - Create an iCloud Container by using [Apple Developer Portal](https://developer.apple.com/account/resources/identifiers/list/cloudContainer)
+   - Set the Identifier to `iCloud.com.example.bitwarden`
+4. Create 4 App Ids
+   - Create App Ids by using [Apple Developer Portal](https://developer.apple.com/account/resources/identifiers/list)
+   - `com.example.bitwarden` with capabilities:
+        - App Groups: seleect the one we created in ***Step 2***
```
Comment (pavelzw) [inline CONTRIBUTING.md:62]:
```suggestion
        - App Groups: select the one we created in ***Step 2***
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -38,3 +38,218 @@ If you are interested in helping translate the Bitwarden mobile app into another
...
+        - Associated Domains: select the one we created in ***Step 3***
+        - AutoFill Credential Provider
+        - iCloud 
+        - NFC Tag Reading
+        - Push Notifications (no need to configure certificates)
+   - `com.example.bitwarden.autofill` with capabilities:
+        - App Groups: seleect the one we created in step 2
```
Comment (pavelzw) [inline CONTRIBUTING.md:69]:
```suggestion
        - App Groups: select the one we created in step 2
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -38,3 +38,218 @@ If you are interested in helping translate the Bitwarden mobile app into another
...
+        - AutoFill Credential Provider
+        - Push Notifications (no need to configure certificates)
+   - `com.example.bitwarden.find-login-action-extension`
+        - App Groups: seleect the one we created in step 2
```
Comment (pavelzw) [inline CONTRIBUTING.md:73]:
```suggestion
        - App Groups: select the one we created in step 2
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -38,3 +38,218 @@ If you are interested in helping translate the Bitwarden mobile app into another
...
+        - Push Notifications (no need to configure certificates)
+   - `com.example.bitwarden.share-extension`
+        - App Groups: seleect the one we created in step 2
```
Comment (pavelzw) [inline CONTRIBUTING.md:76]:
```suggestion
        - App Groups: select the one we created in step 2
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -38,3 +38,218 @@ If you are interested in helping translate the Bitwarden mobile app into another
...
+3. Create an iCloud Container:
+   - Create an iCloud Container by using [Apple Developer Portal](https://developer.apple.com/account/resources/identifiers/list/cloudContainer)
+   - Set the Identifier to `iCloud.com.example.bitwarden`
+4. Create 4 App Ids
+   - Create App Ids by using [Apple Developer Portal](https://developer.apple.com/account/resources/identifiers/list)
+   - `com.example.bitwarden` with capabilities:
+        - App Groups: seleect the one we created in ***Step 2***
+        - Associated Domains: select the one we created in ***Step 3***
+        - AutoFill Credential Provider
+        - iCloud
```
Comment (pavelzw) [inline CONTRIBUTING.md:65]:
```suggestion
        - Associated Domains
        - AutoFill Credential Provider
        - iCloud: select the one we created in ***Step 3***
```

**Comment by junyuan-z** [comment]
so should I make the change & try to land it into this repo?

**Comment by eliykat** [comment]
We now have a public documentation repo you can submit it to: https://github.com/bitwarden/contributing-docs

Specifically I think it would live somewhere in here: https://github.com/bitwarden/contributing-docs/tree/master/docs/getting-started/clients/mobile/ios

If you open a new PR there and close this one, I can chase up the mobile team for a review. Thanks!

**Pull request closed without merging**